File: app/src/main/java/com/ryanpconnors/art_thief/compare/SortArtWorkActivity.java
package com.ryanpconnors.art_thief.compare;

import android.net.Uri;
import android.support.v4.app.Fragment;

import com.ryanpconnors.art_thief.SingleFragmentActivity;

public class SortArtWorkActivity extends SingleFragmentActivity implements SortArtWorkFragment.OnSortArtworkFragmentInteractionListener {

    public static final String EXTRA_STAR_COUNT = "com.ryanpconnors.artthief.star_count";

    @Override
    protected Fragment createFragment() {
        int starCount = (Integer) getIntent().getSerializableExtra(EXTRA_STAR_COUNT);
        return SortArtWorkFragment.newInstance(starCount);
    }

    @Override
    public void onFragmentInteraction(Uri uri) {
        // TODO:
    }
}

File: app/src/main/java/com/ryanpconnors/art_thief/compare/SortArtWorkActivity.kt
package com.ryanpconnors.art_thief.compare

import android.net.Uri
import android.support.v4.app.Fragment

import com.ryanpconnors.art_thief.SingleFragmentActivity

class SortArtWorkActivity : SingleFragmentActivity(), SortArtWorkFragment.OnSortArtworkFragmentInteractionListener {

    override fun createFragment(): Fragment {
        val starCount = intent.getSerializableExtra(EXTRA_STAR_COUNT) as Int
        return SortArtWorkFragment.newInstance(starCount)
    }

    override fun onFragmentInteraction(uri: Uri) {
        // TODO:
    }

    companion object {

        val EXTRA_STAR_COUNT = "com.ryanpconnors.artthief.star_count"
    }
}
